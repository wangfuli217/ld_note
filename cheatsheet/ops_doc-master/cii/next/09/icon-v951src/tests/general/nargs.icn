#SRC: JCON

#  check return values from args()

#%#%  later: add graphics procedures

procedure main()
   nargs(abs, "abs")
   nargs(acos, "acos")
   nargs(any, "any")
   nargs(args, "args")
   nargs(asin, "asin")
   nargs(atan, "atan")
   nargs(bal, "bal")
   nargs(center, "center")
   nargs(char, "char")
   nargs(close, "close")
   nargs(collect, "collect")
   nargs(copy, "copy")
   nargs(cos, "cos")
   nargs(cset, "cset")
   nargs(delay, "delay")
   nargs(delete, "delete")
   nargs(detab, "detab")
   nargs(display, "display")
   nargs(dtor, "dtor")
   nargs(entab, "entab")
   nargs(errorclear, "errorclear")
   nargs(exit, "exit")
   nargs(exp, "exp")
   nargs(find, "find")
   nargs(flush, "flush")
   nargs(function, "function")
   nargs(get, "get")
   nargs(getenv, "getenv")
   nargs(iand, "iand")
   nargs(icom, "icom")
   nargs(image, "image")
   nargs(insert, "insert")
   nargs(integer, "integer")
   nargs(ior, "ior")
   nargs(ishift, "ishift")
   nargs(ixor, "ixor")
   nargs(key, "key")
   nargs(left, "left")
   nargs(list, "list")
   nargs(log, "log")
   nargs(many, "many")
   nargs(map, "map")
   nargs(match, "match")
   nargs(member, "member")
   nargs(move, "move")
   nargs(name, "name")
   nargs(numeric, "numeric")
#  nargs(open, "open")       # not checked: varies depending on #ifdef Graphics
   nargs(ord, "ord")
   nargs(pop, "pop")
   nargs(pos, "pos")
   nargs(proc, "proc")
   nargs(pull, "pull")
   nargs(push, "push")
   nargs(put, "put")
   nargs(read, "read")
   nargs(reads, "reads")
   nargs(real, "real")
   nargs(remove, "remove")
   nargs(rename, "rename")
   nargs(repl, "repl")
   nargs(reverse, "reverse")
   nargs(right, "right")
   nargs(rtod, "rtod")
   nargs(runerr, "runerr")
   nargs(seek, "seek")
   nargs(seq, "seq")
   nargs(serial, "serial")
   nargs(set, "set")
   nargs(sin, "sin")
   nargs(sort, "sort")
   nargs(sortf, "sortf")
   nargs(sqrt, "sqrt")
   nargs(stop, "stop")
   nargs(string, "string")
   nargs(system, "system")
   nargs(tab, "tab")
   nargs(table, "table")
   nargs(tan, "tan")
   nargs(trim, "trim")
   nargs(type, "type")
   nargs(upto, "upto")
   nargs(variable, "variable")
   nargs(where, "where")
   nargs(write, "write")
   nargs(writes, "writes")
end

procedure nargs(p, s)
   write(right(args(p),3), " ", s)
end
