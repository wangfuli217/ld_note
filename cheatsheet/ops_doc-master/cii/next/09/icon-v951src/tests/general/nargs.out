  1 abs
  1 acos
  4 any
  1 args
  1 asin
  2 atan
  6 bal
  3 center
  1 char
  1 close
  2 collect
  1 copy
  1 cos
  1 cset
  1 delay
  2 delete
 -2 detab
  2 display
  1 dtor
 -2 entab
  0 errorclear
  1 exit
  1 exp
  4 find
  1 flush
  0 function
  1 get
  1 getenv
  2 iand
  1 icom
  1 image
  3 insert
  1 integer
  2 ior
  2 ishift
  2 ixor
  1 key
  3 left
  2 list
  2 log
  4 many
  3 map
  4 match
  2 member
  1 move
  1 name
  1 numeric
  1 ord
  1 pop
  1 pos
  2 proc
  1 pull
 -2 push
 -2 put
  1 read
  2 reads
  1 real
  1 remove
  2 rename
  2 repl
  1 reverse
  3 right
  1 rtod
 -2 runerr
  2 seek
  2 seq
  1 serial
  1 set
  1 sin
  2 sort
  2 sortf
  1 sqrt
 -1 stop
  1 string
  1 system
  1 tab
  1 table
  1 tan
  2 trim
  1 type
  4 upto
  1 variable
  1 where
 -1 write
 -1 writes
